Mapbox/MapboxMapsStyle/Types/Color.swift

```diff
@@ -1,48 +1,34 @@
 import UIKit
 
-public typealias ColorRepresentable = String
-extension ColorRepresentable {
+/// Container to represent `UIColor`for use by the map renderer
+public struct ColorRepresentable: Codable, Equatable {
+
+    /// String representation of a `UIColor` used by the renderer
+    public let colorRepresentation: String?
 
     /// Create a string representation of a `UIColor`
-    public init(color: UIColor) {
+    /// - Parameter color: A `UIColor` instance in the sRGB color space
+    /// - Returns: Initializes a `ColorRepresentable` instance if the `color` is in sRGB color space. Returns `nil` otherwise.
+    public init?(color: UIColor) {
         var red: CGFloat = 0.0
         var green: CGFloat = 0.0
         var blue: CGFloat = 0.0
         var alpha: CGFloat = 0.0
-        color.getRed(&red, green: &green, blue: &blue, alpha: &alpha)
-        self = "rgba(\(red * 255.0), \(green * 255.0), \(blue * 255.0), \(alpha))"
+        if color.getRed(&red, green: &green, blue: &blue, alpha: &alpha) {
+            self.colorRepresentation = "rgba(\(red * 255.0), \(green * 255.0), \(blue * 255.0), \(alpha))"
```

Comment:
On iOS 10 and later, `getRed(_:green:blue:alpha:)` uses the extended sRGB color space for the RGB values. This means that the returned values may be less than 0 and greater than 1.

How should we handle such cases?

Reply:
ah, good point! Considering that the renderer only supports the sRGB color space, feels like we should just return nil / fail in that situation.

Reply:
I wonder if there's a standard way to clamp to [0, 1]. Presumably the system does this when it encounters an extended range color that it needs to render on a display that only supports sRGB.

Reply:
For future reference: https://developer.apple.com/documentation/coregraphics/cgcolor/1455493-converted